plugins/outputs/sematext/sematext.go

```diff
@@ -0,0 +1,150 @@
+)
+
+// Sematext struct contains configuration read from Telegraf config and a few runtime objects.
+// We'll use one separate instance of Telegraf for each monitored service. Therefore, token for particular service
+// will be configured on Sematext output level
+type Sematext struct {
+	ReceiverUrl string          `toml:"receiverUrl"`
+	Token       string          `toml:"token"`
+	ProxyServer string          `toml:"proxyServer"`
+	Username    string          `toml:"username"`
+	Password    string          `toml:"password"`
+	Log         telegraf.Logger `toml:"-"`
+
+	metricsUrl   string
+	sender       *sender.Sender
+	senderConfig *sender.Config
+	serializer   serializer.MetricSerializer
+	processors   []processors.Processor
+}
+
+// TODO add real sample config
+const sampleConfig = `
+  ## Sample config for Sematext output
+`
+
+// Connect is no-op for Sematext output plugin, everything was set up before in Init() method
+func (s *Sematext) Connect() error {
+	return nil
+}
+
+// Close Closes the Sematext output
+func (s *Sematext) Close() error {
+	s.sender.Close()
+	return nil
+}
+
+// SampleConfig Returns a sample configuration for the Sematext output
+func (s *Sematext) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the description for the Sematext output
+func (s *Sematext) Description() string {
+	return "Use telegraf to send metrics to Sematext"
+}
+
+// Init performs full initialization of Sematext output
+func (s *Sematext) Init() error {
+	if len(s.Token) == 0 {
+		s.Log.Error("'token' is a required field for Sematext output")
+		return fmt.Errorf("'token' is a required field for Sematext output")
+	}
+	if len(s.ReceiverUrl) == 0 {
+		s.ReceiverUrl = defaultSematextMetricsReceiverUrl
+	}
+
+	proxyURL, err := url.Parse(s.ProxyServer)
+	if err != nil {
+		s.Log.Errorf("invalid ProxyServer: %v", err)
+		return fmt.Errorf("invalid ProxyServer: %v", err)
+	}
+	s.senderConfig = &sender.Config{
+		ProxyURL: proxyURL,
+		Username: s.Username,
+		Password: s.Password,
+	}
+	s.sender = sender.NewSender(s.senderConfig)
+	s.metricsUrl = s.ReceiverUrl + "/write?db=metrics"
+
+	s.initProcessors()
+
+	s.serializer = serializer.NewMetricSerializer()
+
+	s.Log.Infof("Sematext output started with Token=%s, ReceiverUrl=%s, ProxyServer=%s", s.Token, s.ReceiverUrl,
+		s.ProxyServer)
+
+	return nil
+}
+
+// initProcessors instantiates all metric processors that will be used to prepare metrics/tags for sending to Sematext
+func (s *Sematext) initProcessors() {
+	// add more processors as they are implemented
+	s.processors = []processors.Processor{
+		&processors.Token{
+			Token: s.Token,
+		},
+	}
+}
+
+// Write sends metrics to Sematext backend and handles the response
+func (s *Sematext) Write(metrics []telegraf.Metric) error {
+	for _, p := range s.processors {
+		err := p.Process(metrics)
+
+		if err != nil {
+			s.Log.Errorf("can't process metrics in Sematext output, error : %s", err.Error())
+			return err
```

Comment:
should we return from this method if any of the processors fail?

Reply:
It seems we should probably just skip particular metric. We shouldn't return an error, because it is a signal for Telegraf to retry the batch. But if processor failed, there is no point in retrying and we'll just fill the batch pipeline. I'll rework this.